app/composables/useContentList.ts

```
@@ -14,19 +14,52 @@ interface FetchResponse {
   const errorState = useState<string | null>(resolveKey('error'), () => null)
   const initializedState = useState<boolean>(resolveKey('initialized'), () => false)
 
+  // Migration: Check for old key format and migrate data if present
+  // Old format: `default:${pageSize}` (without version or active/archived suffix)
+  // Only migrate when using default key (not custom stateKey) and new state is empty
+  if (!options?.stateKey && !initializedState.value && itemsState.value.length === 0) {
+    const oldBaseKey = `default:${pageSize}`
+    const oldResolveKey = (segment: string) => `content-list:${oldBaseKey}:${segment}`
+
+    // Check if old state exists
+    const oldItemsState = useState<ContentListItem[]>(oldResolveKey('items'))
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)